### DISCUSSION TITLE
Obtaining curve equation

### DISCUSSION THREAD

**Discussion by nahroge** [ChevronOne/point_projection#1] [Q&A] [answered]
Hi @ChevronOne and/or other folks who are using this repo, 

I would like to have the equation of the created curve. However, I got a bit confused. Could you show me how I can get the equation?

I hope you will see my question and thanks in advance.

**Comment by ChevronOne** [comment, marked as answer]
Hi, the fitting function fits cubic-Bézier curves, it returns four control points (_P0_ to _P3_) for each curve and by which the equation of the curve _C_ is then defined as follows: 
https://bit.ly/39LnG93 "C(t) = \sum_{i=0}^{n}\binom{n}{i} (1-t)^{n-i}t^{i}P_i"
 where <img src="https://bit.ly/39KkIRS" align="center" border="0" alt="\binom{n}{i} = \frac{n!}{i!(n-i)!}" width="126" height="46" /> and the curve is defined only in <img src="https://bit.ly/3wKJ7ip" align="center" border="0" alt="t \in [0,1]" width="76" height="18" />. For cubic curves _n=3_.
This can simplified further to https://bit.ly/3wFIBBZ "C(t)=(1-t)^{3}P_{0}+3(1-t)^{2}tP_{1}+3(1-t)t^{2}P_{2}+t^{3}P_{3}, 0\leq t\leq 1"
"if the equations above did not render properly, you can copy-paste them in a proper _latex_ renderer".
Also if you're not familiar with Bézier or parametric curves in general, here's a [quick introduction](https://web.mit.edu/hyperbook/Patrikalakis-Maekawa-Cho/node12.html). The [Wikipedia article](https://en.wikipedia.org/wiki/B%C3%A9zier_curve) on the subject is pretty explanatory too.

You can get the four control points of a single curve by writing something like this:  `auto stat = ppl::LERPer::fitSingle(data, control_points, tolerance);`. The control points will then be stored in the vector `control_points`. You may also need to check the error of the fitting process, which will be given by `stat.ERR`, and see if your application can tolerate it.

More about it in [README.md](https://github.com/ChevronOne/point_projection#readme)

**Reply by nahroge** [reply]
Hi again @ChevronOne, thanks a lot for your explanation and prompt answer!
